lib/pbench/server/api/resources/__init__.py

```diff
@@ -557,46 +732,53 @@ def _dispatch(self, method: Callable, request: Request) -> Response:
-                self.logger.warning("{}", str(e))
-                abort(HTTPStatus.FORBIDDEN, message=str(e))
-            except SchemaError as e:
-                self.logger.warning(
-                    "{}: {} on {!r}", self.__class__.__name__, str(e), json_data
-                )
-                abort(HTTPStatus.BAD_REQUEST, message=str(e))
-            except Exception as e:
-                self.logger.exception("POST unexpected {}: {}", e.__class__.__name__, e)
-                abort(
-                    HTTPStatus.INTERNAL_SERVER_ERROR,
-                    message="INTERNAL ERROR IN VALIDATION",
-                )
 
-            # Automatically authorize the operation only if the API schema has the
-            # "user" key of type USERNAME; otherwise we assume that authorization
-            # is unnecessary, or that the API-specific subclass will take care of
-            # that in preprocess.
-
-            # TODO: This should really be more flexible. (A) we should confirm
-            # that we're choosing a USERNAME type instead of blindly targeting
-            # "user" parameter name; (B) we should allow some other USERNAME
-            # typed parameter. I'm deliberately not making these changes in
-            # order to minimize the scope of this refactoring.
-            if "user" in self.schema:
-                user = json_data.get("user")  # original username, not user ID
-                access = new_data.get("access")  # normalized access policy
-                try:
-                    self._check_authorization(user, access)
-                except UnauthorizedAccess as e:
-                    self.logger.warning("{}", e)
-                    abort(HTTPStatus.FORBIDDEN, message="Not Authorized")
-            return method(new_data, request)
-        else:
+        # We don't accept or process a request payload for GET, or if no
+        # parameter schema is defined
+        if not self.schema or method == self._get:
             return method(json_data, request)
 
+        try:
+            new_data = self.schema.validate(json_data)
+        except UnverifiedUser as e:
+            self.logger.warning("{}", str(e))
+            abort(HTTPStatus.FORBIDDEN, message=str(e))
+        except SchemaError as e:
+            self.logger.warning(
+                "{}: {} on {!r}", self.__class__.__name__, str(e), json_data
+            )
+            abort(HTTPStatus.BAD_REQUEST, message=str(e))
+        except Exception as e:
+            self.logger.exception(
+                "Unexpected validation exception in {}: {}",
+                e.__class__.__name__,
+                str(e),
+            )
+            abort(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                message="INTERNAL ERROR IN VALIDATION",
+            )
+
+        # Automatically authorize the operation only if the API schema has the
+        # "user" key of type USERNAME; otherwise we assume that authorization
```

Comment:
Yes; I didn't make that change, here or in the refactoring, though I thought about it. There's no reason to define a "user" parameter that's not a username... it'd be nice to close that hole but I can't realistically get very excited about it in the context of either of these changes.

Then again, it might not have been much harder to fix than it was to write the TODO, which apparently isn't satisfying you. 😁

Reply:
What's not satisfying me is having a comment describing what the code does when the code doesn't do that...yet.  I mean, having aspirations is fine, but that's kind of what `TODO`'s are for....

So, the comment (if you keep it) should describe what the code does **now**, and the subsequent `TODO` should describe how that's inadequate, or it should point out what the code will eventually be changed to do.  Then, once the code is changed, the `TODO` should be removed and the comment should be updated.

Having the future comment in the present code is going be a problem when they drift into the past....